themes/bootstrap3/js/facets.js

```diff
@@ -116,18 +116,283 @@ VuFind.register('facetList', function FacetList() {
+      // We want to keep only if they are not already in current params
+      return !initialFilteredParams.includes(obj);
+    });
+    let removedParams = initialFilteredParams.filter(function isRemoved(obj) {
+      // If param present in clicked but not initial
+      if (clickedParams.includes(obj) === false) {
+        // If the element was previously removed (in JS)
+        // we remove it from the corresponding array (coming back to initial state)
+        let indexRemoved = globalRemovedParams.indexOf(obj);
+        if (indexRemoved !== -1) {
+          globalRemovedParams.splice(indexRemoved, 1);
+          return false;
+        } else {
+          return true;
+        }
+      } else {
+        // If param in both clicked and initial we don't want it
+        return false;
+      }
+    });
+    if (addedParams.length !== 1 || addedParams[0] !== "") {
+      // We don't concat if there is only one empty element
+      globalAddedParams = globalAddedParams.concat(addedParams);
+    }
+    if (removedParams.length !== 1 || removedParams[0] !== "") {
+      // We don't concat if there is only one empty element
+      globalRemovedParams = globalRemovedParams.concat(removedParams);
+    }
+  }
+
+  function multiFacetsSelectionToggle() {
+    isMultiFacetsSelectionActivated = this.checked;
+    let form = document.querySelector('form#' + dateSelectorId);
+    if (form !== null) {
+      form.querySelector('input[type="submit"]').classList.toggle('hidden');
+    }
+    let buttons = document.getElementsByClassName('apply-filters');
+    for (let i = 0; i < buttons.length; i++) {
+      buttons[i].classList.toggle('hidden');
+    }
+    let checkboxes = document.getElementsByClassName('userSelectionMultiFilters');
+    for (let i = 0; i < checkboxes.length; i++) {
+      checkboxes[i].checked = isMultiFacetsSelectionActivated;
+    }
+  }
+
+  function registerCallbackOnApply(callback) {
+    callbackOnApply = callback;
+  }
+
+  function registerCallbackWhenDeactivated(callback) {
+    callbackWhenDeactivated = callback;
+  }
+
+  function applyClickHandling(context) {
+    let finalContext = (typeof context === "undefined") ? defaultContext : context;
+    finalContext.classList.toggle('multiFacetSelection');
+    finalContext.querySelectorAll('a.facet:not(.narrow-toggle), .facet a').forEach(function addListeners(link) {
+      link.addEventListener('click', function handling(e) {
+        if (isMultiFacetsSelectionActivated === true) {
+          handleClickedFacet(e);
+        } else {
+          callbackWhenDeactivated();
+        }
+      });
+    });
+  }
+
+  function addSwitchAndButton(context) {
+    let elem = document.getElementsByClassName('multi-filters-selection')[0].cloneNode(true);
+    let suffix = Date.now();
+    elem.getElementsByClassName('userSelectionMultiFilters')[0].id += suffix;
+    elem.getElementsByTagName('label')[0].attributes['for'].value += suffix;
```

Comment:
It looks like one of our Javascript tools has a suggestion to use `.for.` instead of `['for'].` here.